### PR TITLE
Fix all doctests

### PR DESCRIPTION
**Pull request by ranjanan**
[ci skip]

How do I check if this build is OK? Can travis be made to run doctests?

### CODE REVIEW

File: doc/devdocs/types.rst [hunk 1]
```diff
@@ -319,9 +319,11 @@ the type, which is an object of type :obj:`TypeName`:
      primary: Array{T,N} <: DenseArray{T,N}
      cache: SimpleVector
        ...
+
      linearcache: SimpleVector
        ...
-     uid: Int64 47
+
+     uid: Int64 -7900426068641098781
```
Comment (tkelman) [inline doc/devdocs/types.rst]: that's an odd number, is this platform-dependent or does it change each run?

**Comment by tkelman** [comment]
> Can travis be made to run doctests?

#15815 - maybe when we transition to the post-sphinx doc world, we can get doctest running in Documenter to have a "lax mode" to ignore line numbers but flag errors on anything more complicated that should be looked at, vs "release mode" that can be more picky.

File: doc/devdocs/reflection.rst [hunk 2]
```diff
@@ -106,10 +106,7 @@ variable assignments:
    julia> expand( :(f() = 1) )
    :(begin
            $(Expr(:method, :f))
-           $(Expr(:method, :f, :((Core.svec)((Core.apply_type)(Tuple,(Core.Typeof)(f)),(Core.svec)())), Toplevel LambdaInfo thunk
-   :(begin  # none, line 1:
-           return 1
-       end), false))
+           $(Expr(:method, :f, :((Core.svec)((Core.apply_type)(Tuple,(Core.Typeof)(f)),(Core.svec)())), Toplevel LambdaInfo thunk, false))
```
Comment (tkelman) [inline doc/devdocs/reflection.rst]: I wonder whether this was intentional

Comment (yuyichao) [inline doc/devdocs/reflection.rst]: This is basically the same issue with https://github.com/JuliaLang/julia/pull/17636 and is kind of annoy.

Comment (yuyichao) [inline doc/devdocs/reflection.rst]: This is basically the same issue as https://github.com/JuliaLang/julia/pull/17636 and is kind of annoy.

Comment (tkelman) [inline doc/devdocs/reflection.rst]: can we track down what made that regress and fix it?

File: doc/devdocs/types.rst (hunk 1 above)
Comment (ranjanan) [inline doc/devdocs/types.rst]: Oh, I copied this weird number from the error trace I got (albeit consistently). I think that's just a random integer (`rand(Int`). Also, I changed the number and ran a few times, and it passed, so I don't think this matter. Would you prefer a less weird number up here? :)

Comment (tkelman) [inline doc/devdocs/types.rst]: I'm just wondering why it was previously given as 47 - should check blame to see who added that, and if they might have been running on Linux. Did you run this on mac or?

Comment (ranjanan) [inline doc/devdocs/types.rst]: I ran this on a Mac.

**Comment by ranjanan** [comment]
I noticed that it's probably inevitable that these doctests fail whenever there are modifications made to the standard library and Base. I encountered failures due to changing line numbers on specific file. For example, `in sqrt(::Int64) at ./math.jl:149` had to be changed to `in sqrt(::Int64) at ./math.jl:154`

File: doc/devdocs/types.rst (hunk 1 above)
Comment (tkelman) [inline doc/devdocs/types.rst]: Let me check out this branch on linux and see if it's consistent. I usually do most of the release preparation from Linux, with the exception of code signing the Windows binaries.

**Comment by tkelman** [comment]
Yeah that's the kind of nuisance failure that's not worth worrying about on master. I usually clean those up between backport releases and it's not a big deal since I only need to do it once a month or so and we rarely backport anything that changes doctest outputs more significantly than line numbers. Since master hopefully won't be using Sphinx for all that much longer, let's defer the discussion about running doctests automatically for later.

**Comment by MichaelHatherly** [comment]
> "lax mode" to ignore line numbers but flag errors on anything more complicated that should be looked at, vs "release mode" that can be more picky.

Sounds reasonable, currently its always in "lax mode", but we could add a stricter mode that won't build if it finds *anything at all* wrong.

File: doc/stdlib/collections.rst [hunk 3]
```diff
@@ -1162,7 +1162,7 @@ Dequeues
 
        julia> deleteat!([6, 5, 4, 3, 2, 1], (2, 2))
        ERROR: ArgumentError: indices must be unique and sorted
-        in deleteat!(::Array{Int64,1}, ::Tuple{Int64,Int64}) at ./array.jl:534
+        in deleteat!(::Array{Int64,1}, ::Tuple{Int64,Int64}) at ./array.jl:537
```
Comment (tkelman) [inline doc/stdlib/collections.rst]: this is in a docstring so would get overwritten by genstdlib, can you change it at the source too?

File: doc/devdocs/types.rst (hunk 1 above)
Comment (tkelman) [inline doc/devdocs/types.rst]: okay I get the same result - I bet this was changed by #17405, so it's intentional

File: doc/stdlib/collections.rst (hunk 3 above)
Comment (ranjanan) [inline doc/stdlib/collections.rst]: Got it. Fixed in the latest commit. Thanks!

File: doc/devdocs/reflection.rst (hunk 2 above)
Comment (tkelman) [inline doc/devdocs/reflection.rst]: x-ref https://github.com/JuliaLang/julia/pull/17518#issuecomment-234987541, looks complicated

**Pull request merged**